src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1214,88 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
             return new Tuple<SqlConnectionInfo, DataTable, ServerConnection>(sqlConnInfo, dt, serverConnection);
         }
 
+        internal async Task HandleAgentNotebooksRequest(AgentNotebooksParams parameters, RequestContext<AgentNotebooksResult> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                try
+                {
+                    var result = new AgentNotebooksResult();
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                        parameters.OwnerUri,
+                        out connInfo);
+
+                    if (connInfo != null)
+                    {
+                        using (SqlConnection connection = new SqlConnection(ConnectionService.BuildConnectionString(connInfo.ConnectionDetails)))
```

Comment:
I think we're generally using the QueryService and ConnectionService to execute SQL rather than directly executing like this.  I think these services provide some abstractions and supporting functionality that's useful.  

Can you look for other examples of executing SQL in the code base and follow those patterns?  Or was this copied from an example in the code?